File: doc-examples/example-java/src/test/java/example/PricingService.java
package example;

import io.micronaut.el.annotation.ELFunction;
import jakarta.inject.Singleton;

@Singleton // <1>
public class PricingService {

    private final double discount = 0.1;

    @ELFunction(prefix = "pricing") // <2>
    public double quote(Book book, int quantity) {
        return book.getUnitPrice() * quantity * (1 - discount);
    }

    @ELFunction(prefix = "pricing") // <3>
    public static String currency() {
        return "EUR";
    }
}

File: doc-examples/example-python/src/test/python/example/PricingService.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Executable
from micronaut.el.annotation import ELFunction

from example.Book import Book


@Singleton  # <1>
class PricingService:

    discount: float = 0.1

    @Executable
    @ELFunction(prefix="pricing")  # <2>
    def quote(self, book: Book, quantity: int) -> float:
        return book.unitPrice * quantity * (1 - self.discount)

    @staticmethod
    @Executable
    @ELFunction(prefix="pricing", name="currency")  # <3>
    def currencyCode() -> str:
        return "EUR"
